lib/src/endpoints/users.dart

```diff
@@ -27,12 +27,30 @@ class Users extends EndpointPaging {
     return Player.fromJson(map);
   }
 
-  Future<Iterable<Track>> recentlyPlayed() async {
-    var jsonString = await _api._get('v1/me/player/recently-played');
+  Future<Iterable<PlayHistory>> recentlyPlayed({int limit, DateTime after, DateTime before}) async {
+    if (after != null && before != null) {
+      throw new Exception('Cannot specify both after and before.');
+    }
+
+    var queryParams = [];
```

Comment:
Please use our existing "tools" for generating the query. We have a [`_buildQuery(...)`](https://github.com/rinukkusu/spotify-dart/blob/61cb1ebd37a037b33766c9164a537cc9ee600b9f/lib/src/endpoints/endpoint_base.dart#L15-L26) method in the base class, which takes a `Map` and produces the query string. It also filters out keys where the values are `null`. For usage you can check [here](https://github.com/rinukkusu/spotify-dart/blob/332f116fc0965fc83ceeee080f64a8e78dfdb4fb/lib/src/endpoints/playlists.dart#L75).
Thanks!